packages/pagination/src/Pagination.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useContext } from 'react';
+import PropTypes from 'prop-types';
+import isFunction from 'lodash.isfunction';
+import { useEffectAsync, useToggle } from '@availity/hooks';
+
+export const PaginationContext = React.createContext();
+
+export const usePagination = () => useContext(PaginationContext);
+
+// todos
+// Add another `useEffect` for only updating the items without calling the function if the `itemsPerPage` prop changes
+
+/**
+ * Pagination Provider that takes in an async function or items and provides a list of items
+ * @param {Function|Object} items - The Items or function to use for pagination
+ * @param {number} itemsPerPage - The items to show per page
+ */
+const Pagination = ({ items: theItems, itemsPerPage, children }) => {
+  const [currentPage, setPage] = useState(1);
+  const [pageData, setPageData] = useState({
+    total: theItems != null ? theItems.totalCount : 0,
+    pages: [],
+    page: {},
+    lower: 0,
+    upper: 0,
+  });
+
+  const [loading, toggleLoading] = useToggle(true);
+
+  useEffectAsync(async () => {
+    if (!loading) {
+      toggleLoading();
+    }
+
+    // If the items is a function then await the resposne in case of async actions
+    const { items, totalCount } = isFunction(theItems)
+      ? await theItems(currentPage, itemsPerPage)
+      : { items: theItems };
+
+    // Get index of item at the start of the currentPage
+    const lower = currentPage === 1 ? 1 : (currentPage - 1) * itemsPerPage + 1;
+
+    // Get the index of the item at the cut-off of the itemPerPage Count
+    const upper =
+      items.length - currentPage * itemsPerPage > 0
+        ? itemsPerPage * currentPage
+        : items.length;
+
+    // Slice that data if it was given from a function since we don't know if its returning all or not
```

Comment:
This comment makes it sounds like you want to slice the data if it was returned from a function `theItems`, but the it ends up slicing only if `theItems` is not a function. Either the comment is misleading or the ternary is backwards.

Reply:
Yeah the idea here was that if the user is providing `theItems` as a function and we are passing the currentPage to it as an arg then the user is in control of the data returned from the function. If the user is passing in an array then its determined to paginate that list and not recalling the function.

Reply:
`Slice that data if it was given from a function`

Reply:
Read over that comment way to quickly, changing that. 👓